modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -333,11 +334,41 @@ protected void setError(String executionId, Throwable t) {
 
     public int getErrorStatus(String executionId) {
         Long status = (Long) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR_STATUS);
-        return status == null ? 0 : status.intValue();
+        if (status != null) {
+            return status.intValue();
+        }
+
+        // check the errors, if the execution is a bulk action
+        AsyncStatus<Serializable> asyncStatus = getAsyncStatus(executionId);
+        if (asyncStatus != null && asyncStatus.hasError()) {
+            return SC_INTERNAL_SERVER_ERROR;
+        }
+
+        return 0;
     }
 
     public String getError(String executionId) {
-        return (String) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR);
+        String error = (String) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR);
+        if (StringUtils.isNotEmpty(error)) {
+            return error;
+        }
+
+        // check the errors, if the execution is a bulk action
+        AsyncStatus<Serializable> asyncStatus = getAsyncStatus(executionId);
+        if (asyncStatus != null) {
+            return asyncStatus.getError();
+        }
+
+        return null;
+    }
+
+    protected AsyncStatus<Serializable> getAsyncStatus(String executionId) {
```

Comment:
I think this should be `AsyncStatus<String>` everywhere to avoid casting later

Reply:
If we want to update that, we have to modify the signature of [AsyncOperationAdapter#AsyncService<Serializable, ?, ?> getAsyncService(String executionId)](https://github.com/nuxeo/nuxeo/blob/00bc07d1f2ff374a703b1babaaa22f9931f8fa1b/modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java#L445) Which will has an impact, I am not sure that the type will be always `String`.

[AsyncService(https://github.com/nuxeo/nuxeo/blob/00bc07d1f2ff374a703b1babaaa22f9931f8fa1b/modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/AsyncService.java#L38)

Reply:
Ok but in the same file all the other calls to `getAsyncService(executionId).getStatus(taskId)` don't need a cast, so I don't know why you have one...

Reply:
ah yes, I think I'll remove this function. And, keep the same logic than the others call.

Thanks,